### PR TITLE
Upgraded FfDL

### PR DESCRIPTION
**Pull request by fplk**
DO NOT MERGE YET

Tested successfully against IBM Cloud except for last commit. Need to retest.
In general, this version should work against DIND Kubernetes and IBM Cloud for both normal training and distributed training.
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by fplk** [comment]
Line 114 in the Makefile is misindented.

Travis fails because it does not know the pull secret which I cannot easily add, since it contains the registry credentials. I assume this https://stackoverflow.com/questions/9338428/using-secret-api-keys-on-travis-ci is the way to go?

**Comment by Tomcli** [comment]
@fplk Yup, I believe Travis will test the code using minikube. If it can't pass the test on minikube, we might need to change it to test with kubeadm-dind.

File: design/design_docs.md [hunk 1]
```diff
@@ -1,33 +0,0 @@
-# FfDL Design Considerations
```
Comment (Tomcli) [inline design/design_docs.md]: We want to keep this design_docs

File: etc/examples/caffe-model/gpu-manifest.yml [hunk 2]
```diff
@@ -2,9 +2,9 @@ name: mnist-caffe-gpu-model
 description: Caffe MNIST model (GPU)
 version: "1.0"
 gpus: 1
-cpus: 2
+cpus: 0.005
```
Comment (Tomcli) [inline etc/examples/caffe-model/gpu-manifest.yml]: The cpus setting is from the old example. we should change it back to 2 for the sample job to be more efficient.

File: etc/examples/tf-model/manifest.yml [hunk 3]
```diff
@@ -8,16 +8,15 @@ memory: 1Gb
...
+      container: dlaas-ci-tf-training-data
     training_results:
-      container: tf_trained_model
+      container: dlaas-ci-trained-results-tf-training-data
     connection:
-      type: s3_datastore
-      auth_url: http://s3.default.svc.cluster.local
-      user_name: test
-      password: test
+      auth_url: https://s3-api.dal-us-geo.objectstorage.service.networklayer.com
```
Comment (Tomcli) [inline etc/examples/tf-model/manifest.yml]: You are exposing your COS credential over here.

File: etc/examples/tf-summary-model/gpu-manifest.yml [hunk 4]
```diff
@@ -2,7 +2,7 @@ name: tf_mnist_with_summaries_gpu_tutorial
 description: Convolutional network model using tensorflow (GPU)
 version: "1.0"
 gpus: 1
-cpus: 2
+cpus: 0.005
```
Comment (Tomcli) [inline etc/examples/tf-summary-model/gpu-manifest.yml]: Please revert the changes from this example.

**Comment by Tomcli** [comment]
Also please removed the `docker-machine-driver-hyperkit` on the main directory. I believe that's just a driver for Minikube.

**Comment by fplk** [comment]
Withdrawn.
